In [ ]:
import pandas as pd


In [3]:
students = ['Alice', 'Jack', 'Molly']


In [4]:
pd.Series(students)

0    Alice
1     Jack
2    Molly
dtype: object

In [5]:
numbers = [1,2,3,4,5]
pd.Series(numbers)

0    1
1    2
2    3
3    4
4    5
dtype: int64

In [8]:
students_scores = {'Alice' : 'Physics',
                   'Jack' : 'Chemistry',
                   'Molly' : 'English'
    }
s = pd.Series(students_scores, index={'Alice','Molly','Sam'})
s

Alice    Physics
Sam          NaN
Molly    English
dtype: object

In [11]:
students_scores = {'Alice' : 'Physics',
                   'Jack' : 'Chemistry',
                   'Molly' : 'English',
                   'Sam' : 'History'
    }
s = pd.Series(students_scores)

In [13]:
s.iloc[3]

'History'

In [14]:
s.loc['Sam']

'History'

In [15]:
s[3]

'History'

In [16]:
s['Molly']

'English'

In [18]:
import numpy as np
numbers = np.random.randint(0,1000,10000)

In [19]:
numbers



array([ 10, 606, 885, ..., 245, 979, 779])

In [20]:
len(numbers)

10000

In [21]:
%%timeit -n 100
total = 0
for number in numbers:
    total += number
total/len(numbers)

655 µs ± 53.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


In [22]:
numbers = pd.Series(numbers)

In [23]:
%%timeit -n 100
total = 0
for number in numbers:
    total += number
total/len(numbers)

872 µs ± 18.6 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


In [24]:
%%timeit -n 100
total = 0
total = np.sum(numbers)
total/len(numbers)

73.7 µs ± 39.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


In [26]:
%%timeit -n 1000
total = 0
for number in numbers:
    total += number
total/len(numbers)

857 µs ± 3.02 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


In [27]:
%%timeit -n 1000
total = 0
total = np.sum(numbers)
total/len(numbers)

40.2 µs ± 3.39 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


In [28]:
numbers.head()

0     10
1    606
2    885
3      3
4    456
dtype: int32

In [29]:
numbers += 2
numbers.head()

0     12
1    608
2    887
3      5
4    458
dtype: int32

In [31]:
%%timeit -n 10
s = pd.Series(np.random.randint(0,1000,1000))
for label, value in s.iteritems():
    s.loc[label] = value+2
    

28.5 ms ± 348 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


In [32]:
%%timeit -n 10
s = pd.Series(np.random.randint(0,1000,1000))
s+=2

186 µs ± 53.2 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


*NOTE* 0.028 sec ~ 28.5 ms
0.000186 sec ~ 186 microsec

# Let's start with DataFrames

In [38]:
record1 = pd.Series({'Name':'Kelly',
                     'Subject':'History',
                     'Score':90})
record2 = pd.Series({'Name':'Ashley',
                     'Subject':'Arts',
                     'Score':85})
record3 = pd.Series({'Name':'Molly',
                     'Subject':'Chemistry',
                     'Score':85})


In [39]:
record1


Name         Kelly
Subject    History
Score           90
dtype: object

In [42]:
df = pd.DataFrame([record1,record2,record3],index=['student1','student2','student3'])

In [43]:
df

Name    Subject  Score
student1   Kelly    History     90
student2  Ashley       Arts     85
student3   Molly  Chemistry     85

In [51]:
record4 = pd.Series({'Name':'Jack',
                     'Subject':'History',
                     'Score':90})
record5 = pd.Series({'Name':'Jen',
                     'Subject':'Arts',
                     'Score':85})
record6 = pd.Series({'Name':'Mark',
                     'Subject':'Chemistry',
                     'Score':85})

In [54]:
df = pd.DataFrame([record4,record5,record6],index=['student1','student2','student1'])
df

Name    Subject  Score
student1  Jack    History     90
student2   Jen       Arts     85
student1  Mark  Chemistry     85

In [58]:
df.loc['student1', 'Name']

student1    Jack
student1    Mark
Name: Name, dtype: object

In [59]:
df.T

student1 student2   student1
Name        Jack      Jen       Mark
Subject  History     Arts  Chemistry
Score         90       85         85

In [66]:
df.iloc[1]

Name        Jen
Subject    Arts
Score        85
Name: student2, dtype: object

In [67]:
df

Name    Subject  Score
student1  Jack    History     90
student2   Jen       Arts     85
student1  Mark  Chemistry     85

In [74]:
df.loc[:,['Subject','Score']]

Subject  Score
student1    History     90
student2       Arts     85
student1  Chemistry     85

In [76]:
  df = pd.read_csv('datasets/Admission_Predict.csv')

In [78]:
df.head()

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0           1        337          118                  4  4.5   4.5  9.65   
1           2        324          107                  4  4.0   4.5  8.87   
2           3        316          104                  3  3.0   3.5  8.00   
3           4        322          110                  3  3.5   2.5  8.67   
4           5        314          103                  2  2.0   3.0  8.21   

   Research  Chance of Admit   
0         1              0.92  
1         1              0.76  
2         1              0.72  
3         1              0.80  
4         0              0.65

In [79]:
newdf = df.copy()
newdf

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0             1        337          118                  4  4.5   4.5  9.65   
1             2        324          107                  4  4.0   4.5  8.87   
2             3        316          104                  3  3.0   3.5  8.00   
3             4        322          110                  3  3.5   2.5  8.67   
4             5        314          103                  2  2.0   3.0  8.21   
..          ...        ...          ...                ...  ...   ...   ...   
395         396        324          110                  3  3.5   3.5  9.04   
396         397        325          107                  3  3.0   3.5  9.11   
397         398        330          116                  4  5.0   4.5  9.45   
398         399        312          103                  3  3.5   4.0  8.78   
399         400        333          117                  4  5.0   4.0  9.66   

     Research  Chance of Admit   
0           1              0.92  
1           1              0.76  
2           1              0.72  
3           1              0.80  
4           0              0.65  
..        ...               ...  
395         1              0.82  
396         1              0.84  
397         1              0.91  
398         0              0.67  
399         1              0.95  

[400 rows x 9 columns]

In [82]:
newdf = newdf.rename(columns={'SOP':'Statement of Purpose',
                             'LOR' :'Letter of Recommendation'})

In [83]:
newdf

Serial No.  GRE Score  TOEFL Score  University Rating  \
0             1        337          118                  4   
1             2        324          107                  4   
2             3        316          104                  3   
3             4        322          110                  3   
4             5        314          103                  2   
..          ...        ...          ...                ...   
395         396        324          110                  3   
396         397        325          107                  3   
397         398        330          116                  4   
398         399        312          103                  3   
399         400        333          117                  4   

     Statement of Purpose  LOR   CGPA  Research  Chance of Admit   
0                     4.5   4.5  9.65         1              0.92  
1                     4.0   4.5  8.87         1              0.76  
2                     3.0   3.5  8.00         1              0.72  
3                     3.5   2.5  8.67         1              0.80  
4                     2.0   3.0  8.21         0              0.65  
..                    ...   ...   ...       ...               ...  
395                   3.5   3.5  9.04         1              0.82  
396                   3.0   3.5  9.11         1              0.84  
397                   5.0   4.5  9.45         1              0.91  
398                   3.5   4.0  8.78         0              0.67  
399                   5.0   4.0  9.66         1              0.95  

[400 rows x 9 columns]

In [85]:
newdf.columns



Index(['Serial No.', 'GRE Score', 'TOEFL Score', 'University Rating',
       'Statement of Purpose', 'LOR ', 'CGPA', 'Research', 'Chance of Admit '],
      dtype='object')

In [86]:
newdf=newdf.rename(columns={'LOR ':'Letter of Recommendation'})

In [87]:
newdf

Serial No.  GRE Score  TOEFL Score  University Rating  \
0             1        337          118                  4   
1             2        324          107                  4   
2             3        316          104                  3   
3             4        322          110                  3   
4             5        314          103                  2   
..          ...        ...          ...                ...   
395         396        324          110                  3   
396         397        325          107                  3   
397         398        330          116                  4   
398         399        312          103                  3   
399         400        333          117                  4   

     Statement of Purpose  Letter of Recommendation  CGPA  Research  \
0                     4.5                       4.5  9.65         1   
1                     4.0                       4.5  8.87         1   
2                     3.0                       3.5  8.00         1   
3                     3.5                       2.5  8.67         1   
4                     2.0                       3.0  8.21         0   
..                    ...                       ...   ...       ...   
395                   3.5                       3.5  9.04         1   
396                   3.0                       3.5  9.11         1   
397                   5.0                       4.5  9.45         1   
398                   3.5                       4.0  8.78         0   
399                   5.0                       4.0  9.66         1   

     Chance of Admit   
0                0.92  
1                0.76  
2                0.72  
3                0.80  
4                0.65  
..                ...  
395              0.82  
396              0.84  
397              0.91  
398              0.67  
399              0.95  

[400 rows x 9 columns]

In [89]:
newdf.columns

Index(['Serial No.', 'GRE Score', 'TOEFL Score', 'University Rating',
       'Statement of Purpose', 'Letter of Recommendation', 'CGPA', 'Research',
       'Chance of Admit '],
      dtype='object')

In [92]:
cols = list(newdf.columns)
cols = [x.lower().strip() for x in cols]
newdf.columns = cols
newdf.columns

Index(['serial no.', 'gre score', 'toefl score', 'university rating',
       'statement of purpose', 'letter of recommendation', 'cgpa', 'research',
       'chance of admit'],
      dtype='object')

In [93]:
df


Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0             1        337          118                  4  4.5   4.5  9.65   
1             2        324          107                  4  4.0   4.5  8.87   
2             3        316          104                  3  3.0   3.5  8.00   
3             4        322          110                  3  3.5   2.5  8.67   
4             5        314          103                  2  2.0   3.0  8.21   
..          ...        ...          ...                ...  ...   ...   ...   
395         396        324          110                  3  3.5   3.5  9.04   
396         397        325          107                  3  3.0   3.5  9.11   
397         398        330          116                  4  5.0   4.5  9.45   
398         399        312          103                  3  3.5   4.0  8.78   
399         400        333          117                  4  5.0   4.0  9.66   

     Research  Chance of Admit   
0           1              0.92  
1           1              0.76  
2           1              0.72  
3           1              0.80  
4           0              0.65  
..        ...               ...  
395         1              0.82  
396         1              0.84  
397         1              0.91  
398         0              0.67  
399         1              0.95  

[400 rows x 9 columns]

In [94]:
newdf

serial no.  gre score  toefl score  university rating  \
0             1        337          118                  4   
1             2        324          107                  4   
2             3        316          104                  3   
3             4        322          110                  3   
4             5        314          103                  2   
..          ...        ...          ...                ...   
395         396        324          110                  3   
396         397        325          107                  3   
397         398        330          116                  4   
398         399        312          103                  3   
399         400        333          117                  4   

     statement of purpose  letter of recommendation  cgpa  research  \
0                     4.5                       4.5  9.65         1   
1                     4.0                       4.5  8.87         1   
2                     3.0                       3.5  8.00         1   
3                     3.5                       2.5  8.67         1   
4                     2.0                       3.0  8.21         0   
..                    ...                       ...   ...       ...   
395                   3.5                       3.5  9.04         1   
396                   3.0                       3.5  9.11         1   
397                   5.0                       4.5  9.45         1   
398                   3.5                       4.0  8.78         0   
399                   5.0                       4.0  9.66         1   

     chance of admit  
0               0.92  
1               0.76  
2               0.72  
3               0.80  
4               0.65  
..               ...  
395             0.82  
396             0.84  
397             0.91  
398             0.67  
399             0.95  

[400 rows x 9 columns]

In [106]:
newdf[['gre score', 'cgpa']]

gre score  cgpa
0          337  9.65
1          324  8.87
2          316  8.00
3          322  8.67
4          314  8.21
..         ...   ...
395        324  9.04
396        325  9.11
397        330  9.45
398        312  8.78
399        333  9.66

[400 rows x 2 columns]

In [1]:
newdf[newdf['gre score']>320].head()

NameError: name 'newdf' is not defined

In [8]:
import pandas as pd
student_df = pd.DataFrame([{'Name':'Kelly', 'Class':'Law'},
                          {'Name':'Ashley', 'Class':'Arts'},
                          {'Name':'Molly', 'Class':'Management'}])
student_df = student_df.set_index('Name')

In [9]:
student_df

Class
Name              
Kelly          Law
Ashley        Arts
Molly   Management

In [11]:
staff_df = pd.DataFrame([{'Name':'Kelly', 'Role':'Dean'},
                          {'Name':'Ashley', 'Role':'HR'},
                          {'Name':'James', 'Role':'Grader'}])
staff_df = staff_df.set_index('Name')

staff_df

Role
Name          
Kelly     Dean
Ashley      HR
James   Grader

In [14]:
pd.merge(staff_df,student_df,how='outer',left_index=True,right_index=True) #Union

Role       Class
Name                      
Ashley      HR        Arts
James   Grader         NaN
Kelly     Dean         Law
Molly      NaN  Management

In [15]:
pd.merge(staff_df,student_df,how='inner',left_index=True,right_index=True) #Intersection

Role Class
Name              
Kelly   Dean   Law
Ashley    HR  Arts

In [16]:
pd.merge(staff_df,student_df,how='left',left_index=True,right_index=True) 

Role Class
Name                
Kelly     Dean   Law
Ashley      HR  Arts
James   Grader   NaN

In [17]:
pd.merge(staff_df,student_df,how='right',left_index=True,right_index=True) #Union

Role       Class
Name                    
Kelly   Dean         Law
Ashley    HR        Arts
Molly    NaN  Management

In [18]:
staff_df.reset_index()

Name    Role
0   Kelly    Dean
1  Ashley      HR
2   James  Grader

In [20]:
student_df.reset_index()

Name       Class
0   Kelly         Law
1  Ashley        Arts
2   Molly  Management

In [25]:
staff_df = pd.DataFrame([{'Name':'Kelly', 'Role':'Dean', 'Location':'Fifth Avenue'},
                          {'Name':'Ashley', 'Role':'HR', 'Location':'Center Cross'},
                          {'Name':'James', 'Role':'Grader', 'Location':'Third Street'}])
student_df = pd.DataFrame([{'Name':'Kelly', 'Class':'Law', 'Location':'Fifth Avenue'},
                          {'Name':'Ashley', 'Class':'Arts', 'Location':'Center Cross'},
                          {'Name':'Molly', 'Class':'Management', 'Location':'Church Street'}])
print(staff_df)
print(student_df)

     Name    Role      Location
0   Kelly    Dean  Fifth Avenue
1  Ashley      HR  Center Cross
2   James  Grader  Third Street
     Name       Class       Location
0   Kelly         Law   Fifth Avenue
1  Ashley        Arts   Center Cross
2   Molly  Management  Church Street


In [28]:
pd.merge(staff_df,student_df,how='outer', on='Name') #location_x refers to left and location_y refers to right

Name    Role    Location_x       Class     Location_y
0   Kelly    Dean  Fifth Avenue         Law   Fifth Avenue
1  Ashley      HR  Center Cross        Arts   Center Cross
2   James  Grader  Third Street         NaN            NaN
3   Molly     NaN           NaN  Management  Church Street

In [37]:
import numpy as np


In [38]:
import timeit

In [53]:
df = pd.read_csv('datasets/census.csv')
df.head()

SUMLEV  REGION  DIVISION  STATE  COUNTY   STNAME         CTYNAME  \
0      40       3         6      1       0  Alabama         Alabama   
1      50       3         6      1       1  Alabama  Autauga County   
2      50       3         6      1       3  Alabama  Baldwin County   
3      50       3         6      1       5  Alabama  Barbour County   
4      50       3         6      1       7  Alabama     Bibb County   

   CENSUS2010POP  ESTIMATESBASE2010  POPESTIMATE2010  ...  RDOMESTICMIG2011  \
0        4779736            4780127          4785161  ...          0.002295   
1          54571              54571            54660  ...          7.242091   
2         182265             182265           183193  ...         14.832960   
3          27457              27457            27341  ...         -4.728132   
4          22915              22919            22861  ...         -5.527043   

   RDOMESTICMIG2012  RDOMESTICMIG2013  RDOMESTICMIG2014  RDOMESTICMIG2015  \
0         -0.193196          0.381066          0.582002         -0.467369   
1         -2.915927         -3.012349          2.265971         -2.530799   
2         17.647293         21.845705         19.243287         17.197872   
3         -2.500690         -7.056824         -3.904217        -10.543299   
4         -5.068871         -6.201001         -0.177537          0.177258   

   RNETMIG2011  RNETMIG2012  RNETMIG2013  RNETMIG2014  RNETMIG2015  
0     1.030015     0.826644     1.383282     1.724718     0.712594  
1     7.606016    -2.626146    -2.722002     2.592270    -2.187333  
2    15.844176    18.559627    22.727626    20.317142    18.293499  
3    -4.874741    -2.758113    -7.167664    -3.978583   -10.543299  
4    -5.088389    -4.363636    -5.403729     0.754533     1.107861  

[5 rows x 100 columns]

In [54]:
df = df[df['SUMLEV'] == 50]
df.set_index(['STNAME','CTYNAME'])


SUMLEV  REGION  DIVISION  STATE  COUNTY  \
STNAME  CTYNAME                                                      
Alabama Autauga County         50       3         6      1       1   
        Baldwin County         50       3         6      1       3   
        Barbour County         50       3         6      1       5   
        Bibb County            50       3         6      1       7   
        Blount County          50       3         6      1       9   
...                           ...     ...       ...    ...     ...   
Wyoming Sweetwater County      50       4         8     56      37   
        Teton County           50       4         8     56      39   
        Uinta County           50       4         8     56      41   
        Washakie County        50       4         8     56      43   
        Weston County          50       4         8     56      45   

                           CENSUS2010POP  ESTIMATESBASE2010  POPESTIMATE2010  \
STNAME  CTYNAME                                                                
Alabama Autauga County             54571              54571            54660   
        Baldwin County            182265             182265           183193   
        Barbour County             27457              27457            27341   
        Bibb County                22915              22919            22861   
        Blount County              57322              57322            57373   
...                                  ...                ...              ...   
Wyoming Sweetwater County          43806              43806            43593   
        Teton County               21294              21294            21297   
        Uinta County               21118              21118            21102   
        Washakie County             8533               8533             8545   
        Weston County               7208               7208             7181   

                           POPESTIMATE2011  POPESTIMATE2012  ...  \
STNAME  CTYNAME                                              ...   
Alabama Autauga County               55253            55175  ...   
        Baldwin County              186659           190396  ...   
        Barbour County               27226            27159  ...   
        Bibb County                  22733            22642  ...   
        Blount County                57711            57776  ...   
...                                    ...              ...  ...   
Wyoming Sweetwater County            44041            45104  ...   
        Teton County                 21482            21697  ...   
        Uinta County                 20912            20989  ...   
        Washakie County               8469             8443  ...   
        Weston County                 7114             7065  ...   

                           RDOMESTICMIG2011  RDOMESTICMIG2012  \
STNAME  CTYNAME                                                 
Alabama Autauga County             7.242091         -2.915927   
        Baldwin County            14.832960         17.647293   
        Barbour County            -4.728132         -2.500690   
        Bibb County               -5.527043         -5.068871   
        Blount County              1.807375         -1.177622   
...                                     ...               ...   
Wyoming Sweetwater County          1.072643         16.243199   
        Teton County              -1.589565          0.972695   
        Uinta County             -17.755986         -4.916350   
        Washakie County          -11.637475         -0.827815   
        Weston County            -11.752361         -8.040059   

                           RDOMESTICMIG2013  RDOMESTICMIG2014  \
STNAME  CTYNAME                                                 
Alabama Autauga County            -3.012349          2.265971   
        Baldwin County            21.845705         19.243287   
        Barbour County            -7.056824         -3.904217   
        Bibb County

In [55]:
df.groupby('STNAME')

In [56]:
print(df.groupby('STNAME'))

In [57]:
a = df.groupby('STNAME')
a

In [61]:
for group,frame in df.groupby('STNAME'):
    avg = np.average(frame['CENSUS2010POP'])
    print('Counties in state ' + group + ' have an average population of ' + str(avg))

Counties in state Alabama have an average population of 71339.34328358209
Counties in state Alaska have an average population of 24490.724137931036
Counties in state Arizona have an average population of 426134.4666666667
Counties in state Arkansas have an average population of 38878.90666666667
Counties in state California have an average population of 642309.5862068966
Counties in state Colorado have an average population of 78581.1875
Counties in state Connecticut have an average population of 446762.125
Counties in state Delaware have an average population of 299311.3333333333
Counties in state District of Columbia have an average population of 601723.0
Counties in state Florida have an average population of 280616.5671641791
Counties in state Georgia have an average population of 60928.63522012578
Counties in state Hawaii have an average population of 272060.2
Counties in state Idaho have an average population of 35626.86363636364
Counties in state Illinois have an average populat

In [63]:
df.columns

Index(['SUMLEV', 'REGION', 'DIVISION', 'STATE', 'COUNTY', 'STNAME', 'CTYNAME',
       'CENSUS2010POP', 'ESTIMATESBASE2010', 'POPESTIMATE2010',
       'POPESTIMATE2011', 'POPESTIMATE2012', 'POPESTIMATE2013',
       'POPESTIMATE2014', 'POPESTIMATE2015', 'NPOPCHG_2010', 'NPOPCHG_2011',
       'NPOPCHG_2012', 'NPOPCHG_2013', 'NPOPCHG_2014', 'NPOPCHG_2015',
       'BIRTHS2010', 'BIRTHS2011', 'BIRTHS2012', 'BIRTHS2013', 'BIRTHS2014',
       'BIRTHS2015', 'DEATHS2010', 'DEATHS2011', 'DEATHS2012', 'DEATHS2013',
       'DEATHS2014', 'DEATHS2015', 'NATURALINC2010', 'NATURALINC2011',
       'NATURALINC2012', 'NATURALINC2013', 'NATURALINC2014', 'NATURALINC2015',
       'INTERNATIONALMIG2010', 'INTERNATIONALMIG2011', 'INTERNATIONALMIG2012',
       'INTERNATIONALMIG2013', 'INTERNATIONALMIG2014', 'INTERNATIONALMIG2015',
       'DOMESTICMIG2010', 'DOMESTICMIG2011', 'DOMESTICMIG2012',
       'DOMESTICMIG2013', 'DOMESTICMIG2014', 'DOMESTICMIG2015', 'NETMIG2010',
       'NETMIG2011', 'NETMIG2012', 'NETMI

In [64]:
df.head()

SUMLEV  REGION  DIVISION  STATE  COUNTY   STNAME         CTYNAME  \
1      50       3         6      1       1  Alabama  Autauga County   
2      50       3         6      1       3  Alabama  Baldwin County   
3      50       3         6      1       5  Alabama  Barbour County   
4      50       3         6      1       7  Alabama     Bibb County   
5      50       3         6      1       9  Alabama   Blount County   

   CENSUS2010POP  ESTIMATESBASE2010  POPESTIMATE2010  ...  RDOMESTICMIG2011  \
1          54571              54571            54660  ...          7.242091   
2         182265             182265           183193  ...         14.832960   
3          27457              27457            27341  ...         -4.728132   
4          22915              22919            22861  ...         -5.527043   
5          57322              57322            57373  ...          1.807375   

   RDOMESTICMIG2012  RDOMESTICMIG2013  RDOMESTICMIG2014  RDOMESTICMIG2015  \
1         -2.915927         -3.012349          2.265971         -2.530799   
2         17.647293         21.845705         19.243287         17.197872   
3         -2.500690         -7.056824         -3.904217        -10.543299   
4         -5.068871         -6.201001         -0.177537          0.177258   
5         -1.177622         -1.748766         -2.062535         -1.369970   

   RNETMIG2011  RNETMIG2012  RNETMIG2013  RNETMIG2014  RNETMIG2015  
1     7.606016    -2.626146    -2.722002     2.592270    -2.187333  
2    15.844176    18.559627    22.727626    20.317142    18.293499  
3    -4.874741    -2.758113    -7.167664    -3.978583   -10.543299  
4    -5.088389    -4.363636    -5.403729     0.754533     1.107861  
5     1.859511    -0.848580    -1.402476    -1.577232    -0.884411  

[5 rows x 100 columns]

In [4]:
nba = pd.read_csv('datasets/nba.csv')
nba

NameError: name 'pd' is not defined

In [8]:
cd ankan/OneDrive/Documents/Codes/Python/


C:\Users\ankan\OneDrive\Documents\Codes\Python


In [11]:
ls

 Volume in drive C is Windows
 Volume Serial Number is 94C6-6244

 Directory of C:\Users\ankan\OneDrive\Documents\Codes

25-08-2021  16:15    <DIR>          .
25-08-2021  16:15    <DIR>          ..
29-09-2020  10:19            12,288 .row_column_mat.c.swp
25-07-2021  11:25    <DIR>          C  C++
14-05-2021  23:06    <DIR>          dem0
22-09-2021  23:52    <DIR>          Notebooks
18-09-2021  23:26    <DIR>          Python
14-05-2021  23:23    <DIR>          Web
               1 File(s)         12,288 bytes
               7 Dir(s)  162,444,038,144 bytes free


In [12]:
cd Notebooks/


C:\Users\ankan\OneDrive\Documents\Codes\Notebooks


In [14]:
import pandas as pd
nba = pd.read_csv('datasets/nba.csv')
nba

Name            Team  Number Position   Age Height  Weight  \
0    Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   
1      Jae Crowder  Boston Celtics    99.0       SF  25.0    6-6   235.0   
2     John Holland  Boston Celtics    30.0       SG  27.0    6-5   205.0   
3      R.J. Hunter  Boston Celtics    28.0       SG  22.0    6-5   185.0   
4    Jonas Jerebko  Boston Celtics     8.0       PF  29.0   6-10   231.0   
..             ...             ...     ...      ...   ...    ...     ...   
453   Shelvin Mack       Utah Jazz     8.0       PG  26.0    6-3   203.0   
454      Raul Neto       Utah Jazz    25.0       PG  24.0    6-1   179.0   
455   Tibor Pleiss       Utah Jazz    21.0        C  26.0    7-3   256.0   
456    Jeff Withey       Utah Jazz    24.0        C  26.0    7-0   231.0   
457            NaN             NaN     NaN      NaN   NaN    NaN     NaN   

               College     Salary  
0                Texas  7730337.0  
1            Marquette  6796117.0  
2    Boston University        NaN  
3        Georgia State  1148640.0  
4                  NaN  5000000.0  
..                 ...        ...  
453             Butler  2433333.0  
454                NaN   900000.0  
455                NaN  2900000.0  
456             Kansas   947276.0  
457                NaN        NaN  

[458 rows x 9 columns]

In [19]:
dff = nba.groupby(['Team','Position'])

In [21]:
dff.first()

Name  Number   Age Height  Weight  \
Team               Position                                                  
Atlanta Hawks      C               Al Horford    15.0  30.0   6-10   245.0   
                   PF          Kris Humphries    43.0  31.0    6-9   235.0   
                   PG         Dennis Schroder    17.0  22.0    6-1   172.0   
                   SF           Kent Bazemore    24.0  26.0    6-5   201.0   
                   SG        Tim Hardaway Jr.    10.0  24.0    6-6   205.0   
...                                       ...     ...   ...    ...     ...   
Washington Wizards C            Marcin Gortat    13.0  32.0   6-11   240.0   
                   PF             Drew Gooden    90.0  34.0   6-10   250.0   
                   PG          Ramon Sessions     7.0  30.0    6-3   190.0   
                   SF            Jared Dudley     1.0  30.0    6-7   225.0   
                   SG           Alan Anderson     6.0  33.0    6-6   220.0   

                                          College      Salary  
Team               Position                                    
Atlanta Hawks      C                      Florida  12000000.0  
                   PF                   Minnesota   1000000.0  
                   PG                 Wake Forest   1763400.0  
                   SF                Old Dominion   2000000.0  
                   SG                    Michigan   1304520.0  
...                                           ...         ...  
Washington Wizards C         North Carolina State  11217391.0  
                   PF                      Kansas   3300000.0  
                   PG                      Nevada   2170465.0  
                   SF              Boston College   4375000.0  
                   SG              Michigan State   4000000.0  

[149 rows x 7 columns]